game/3dobject.cc

```diff
@@ -31,7 +31,7 @@ struct Face {
 void Object3d::loadWavefrontObj(fs::path const& filepath, float scale) {
 	int linenumber = 0;
 	std::string row;
-	std::ifstream file(filepath.c_str(), std::ios::binary);
+	std::ifstream file(filepath.string(), std::ios::binary);
```

Comment:
Would it be possible to write this so that non-ASCII characters work on Windows? I would expect there to be a way to make fs::path and ifstream talk using wchars when running on Windows. Opening files on Windows just won't work with regular char filenames (in UTF-8 like all other OS).

Reply:
actually,that was one of my questions: is path::string() hadling the windows charset correctly? you seem to say no... so I will probably take the other version of my path where I was using path directly but need more "hack" stuff in the fs.h.
This was the minimalistic patch to compile, as I cannot test on windows, I chose KISS.
I'll push an update let me know if you think it is better.

Reply:
At best it could be using the ANSI codepage currently active on Windows (depends on locale settings) but still cannot use any characters not included in those less than 256 that the codepage can hold. The only way to open all files on Windows is to use wchar filenames, but C++ fstreams should on Windows have overloads taking wchar and I believe that the fs library should be able to talk directly with that (but I myself am not quite up to date as I do no development on Windows nowadays).

Reply:
Okay, looking at std::filesystem reference, you need `filepath.native()` instead of `filepath.string()`

Reply:
now I use path directly, is that ok?

Reply:
... or just pass the filepath object directly and you'll get implicit conversion to wstring or string depending on platform.